### ISSUE TITLE
[Question] Runtime configuration

### ISSUE THREAD

**Issue by sweco-semtne** [spreedly/kaffe#31]
Hi, 
Thank you for a great library. 
Is it possible to run Kaffe with Kafka endpoints set at runtime and not att compile time? 

I couldn't find any documentation on this?

Something like below (which does not compile)

```
config :kaffe,
  producer: [
    endpoints: [{:system, "KAFKA_SERVER", "defaultname"}, 9092], # [hostname: port]
    topics: ["topicname"],
    # optional
    partition_strategy: :md5
  ]
```

**Comment by sdball** [comment]
Hello! Yes that's entirely possible. The easiest approach would be to set `heroku_kafka_env: true` in the configuration and then ensure that `KAFKA_URL` is set as an ENV variable at runtime. The other Heroku compatible Kafka ENV variables for SSL are optional.

**Comment by rwdaigle** [comment]
@sdball is this conflating Heroku concerns w/ env-var concerns?

**Comment by sdball** [comment]
I don't think so. The flag is for declaring the app is behaving in a Heroku-compatible manner. Kaffe's entire existence is designed around making talking to Heroku Kafka easy.

**Comment by rwdaigle** [comment]
I think I disagree, but can also see it's orthogonal to @sweco-semtne's questions so will take it offline 😄

**Comment by rawkode** [comment]
@sweco-semtne I wrote a library for run-time configuration for Kubernetes and Swarm (though environment vars are supported too)

https://github.com/GT8Online/weave

This is my handler for such a use-case:
```
  def handle_configuration("kafka_hosts", host_string) do
    brokers = host_string
    |> String.split(",")
    |> Enum.map(fn(uri) ->
      [host, port] = String.split(uri, ":")

      Keyword.put([], String.to_atom(host), String.to_integer(port))
    end)
    |> List.flatten()

    {:kaffe, :consumer, [endpoints: brokers]}
  end

  def handle_configuration("kafka_topic", topic) do
    {:kaffe, :consumer, [topics: [topic]]}
  end
```

**Comment by sweco-semtne** [comment]
Thanks! It looks like it is exactly what I'm looking for.

**Comment by sdball** [comment]
Wow, that looks fantastic @rawkode! Would you mind writing up that runtime configuration approach as a PR for the readme? Perhaps another section `Runtime Configuration`?

**Comment by rawkode** [comment]
Yeah, happy to @sdball. No problem :+1:

**Comment by rawkode** [comment]
@sweco-semtne I've made it easier to configure a consumer and producer with a single `handle_configuration/1`:

https://github.com/GT8Online/weave/pull/27/files

**Comment by rawkode** [comment]
I believe this could be closed now, @sweco-semtne ?

Hopefully you got everything working :+1:

**Comment by davidsantoso** [comment]
Looks like this issue has been resolved with #34. Closing due to inactivity.
